virttest/qemu_vm.py

```diff
@@ -1897,6 +1900,48 @@ def sort_key(dev):
                               pci_bus=pci_bus)
                 iov += 1
 
+        def get_cid(cid):
+            """ Get an unused guest cid from system """
+            while cid:
+                cid_c = struct.pack('L', cid)
+                try:
+                    fcntl.ioctl(
+                        vsock_fd, arch.VHOST_VSOCK_SET_GUEST_CID, cid_c)
+                except IOError as e:
+                    if e.errno == errno.EADDRINUSE:
+                        cid += 1
+                        continue
+                    else:
+                        raise e
+                else:
+                    return cid
+
+        # Add vsock device, cid 0-2 are reserved by system
+        vsocks = params.objects('vsocks')
+        if vsocks:
+            vsock_path = "/dev/vhost-vsock"
+            if not os.path.exists(vsock_path):
+                logging.info("vsock module was not loaded, loading it...")
+                loaded = linux_modules.load_module('vsock')
+                loaded &= linux_modules.load_module('vhost_vsock')
+                if not loaded:
+                    raise exceptions.TestError(
+                        "Failed on loading module vsock or vhost_vsock.")
+            vsock_fd = os.open(vsock_path, os.O_RDWR)
+            min_cid = 3
+            for vsock in vsocks:
+                guest_cid = get_cid(min_cid)
+                vsock_params = {"id": vsock, "guest-cid": guest_cid}
+                if '-mmio:' in params.get('machine_type'):
```

Comment:
For both `arm64-mmio` and `riscv64-mmio` ?

Reply:
@yanan-fu I think so, all mmio arch use `-device` instead of `-pci` suffix, that's all same with other virtio device handling. e.g. `virtio-net-device`